### PR TITLE
Added scheme fee to settlement

### PR DESCRIPTION
**Pull request by nilssonemma**

### CODE REVIEW

File: Settlement/Transaction.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export namespace Transaction {
 			isoly.CountryCode.Alpha2.is(value.area) &&
 			isoly.Date.is(value.created) &&
 			typeof value.gross == "number" &&
-			typeof value.fee == "number" &&
+			(typeof value.fee == "number" || (typeof value.fee.scheme == "number" && typeof value.fee.total == "number")) &&
```
Comment (florianebmeier) [inline Settlement/Transaction.ts:32]: Isnt this missing the typeof value.fee == "object" check?

File: Settlement/index.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export namespace Settlement {
 			(value.reserve.payout == undefined || isoly.Date.is(value.reserve.payout)) &&
 			isoly.Date.is(value.created) &&
 			typeof value.gross == "number" &&
-			typeof value.fee == "number" &&
+			(typeof value.fee == "number" || (typeof value.fee.scheme == "number" && typeof value.fee.total)) &&
```
Comment (florianebmeier) [inline Settlement/index.ts:37]: See above

File: Settlement/index.ts [hunk 3]
```diff
@@ -49,9 +49,18 @@ export namespace Settlement {
...
+			result += `"${value.reference}","${value.merchant}","${value.period.start}","${value.period.end}","${
+				value.payout
+			}","${value.reserve?.amount}","${value.reserve?.payout}","${value.created}","${value.gross}","${
+				typeof value.fee == "number" ? value.fee : value.fee.total
+			}","${value.net}","${value.currency}"\r\n`
 			result += value.transactions.length > 0 ? SettlementTransaction.toCsv(value.transactions) : ""
 		}
 		return result
 	}
+	export function toCustomer(settlements: Settlement[]): Settlement[] {
```
Comment (florianebmeier) [inline Settlement/index.ts:61]: i think we want the function to be on the settlement itself so if we have a use case where we only return a single settlement, we can still use the .toCustomer function

Comment (florianebmeier) [reply Settlement/index.ts:61]: Also this is missing that you have to use .toCustomer function on all of the transactions in the transactions field

File: Settlement/Transaction.ts [hunk 4]
```diff
@@ -39,7 +39,17 @@ export namespace Transaction {
...
-			result += `"${value.authorization}","${value.reference}","${value.type}","${value.card}","${value.scheme}","${value.area}","${value.created}","${value.gross}","${value.fee}","${value.net}","${value.reserve?.amount}","${value.reserve?.amount}"\r\n`
+			result += `"${value.authorization}","${value.reference}","${value.type}","${value.card}","${value.scheme}","${
+				value.area
+			}","${value.created}","${value.gross}","${typeof value.fee == "number" ? value.fee : value.fee.total}","${
+				value.net
+			}","${value.reserve?.amount}","${value.reserve?.amount}"\r\n`
 		return result
 	}
+
+	export function toCustomer(transactions: Transaction[]): Transaction[] {
```
Comment (florianebmeier) [inline Settlement/Transaction.ts:50]: Should also be possible for a single transaction, not only for an array.

**Review by florianebmeier — CHANGES_REQUESTED** [review]

**Review by florianebmeier — DISMISSED** [review]

**Review by robinnedergard — DISMISSED** [review]

**Review by robinnedergard — APPROVED** [review]

**Review by florianebmeier — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**